File: lab/python/textos/encontrar_palavras_especificas_com_regex.py
import re


def encontrar_palavras_especificas(string, palavra):
    # rf"" permite interpolar variáveis dentro de regex
    resultado = re.findall(rf"\b{palavra}\w*", string, flags=re.IGNORECASE)
    return resultado


frase = "Eu gosto de programação e projetos produtivos"
resultado = encontrar_palavras_especificas(frase, "pro")
print(resultado)

File: lab/javascript/textos/encontrar_palavras_especificas_com_regex.js
function encontrar_palavras_especificas(string, palavra) {
  // Cria uma regex dinâmica que encontra palavras que começam com "palavra"
  const regex = new RegExp(`\\b${palavra}\\w*`, "gi");
  const resultado = string.match(regex);
  return resultado || [];
}

// -------------------- Teste --------------------
const frase = "Eu gosto de programação e projetos produtivos";
const resultado = encontrar_palavras_especificas(frase, "pro");
console.log(resultado);
